Derive Jump movement

Aioros opened the pull request:
This might be more opinionated than I'm comfortable with, let me know what you think. The safer part of this PR is a correction of the movement cost for Jump to bring it in line with the rules (as described in #6410).

Then, to evaluate the ruler's styling, instead of hardcoding a specific behavior for `jump` I'm proposing a callback configuration property called `deriveSpeed`, which would allow defining a movement type with its own custom calculation for the base speed used by the ruler. This works well enough for Jump, where green (corresponding to half the strength score) would be ok, and yellow (full strength score) would suggest that the movement might have other requirements (the 10ft run-up). But it's definitely possible that this is too specific to be actually useful.

Fyorl reviewed (commented):
No objection to these changes. I think a more forward-looking solution would be to derive a `movement.jump` value from the strength score (after AEs have applied). In v14 you can then target this field with the `final` AE phase allowing for effects that increase jump distance.

Aioros committed:
Correct jump movement cost

Aioros committed:
Additional movement type config properties

- `deriveSpeed` function property, used to evaluate ruler segment styling
- `hidden` optional property, to allow hiding a secondary movement type like jump from actor sheets.

Aioros committed:
Configure jump movement

- derived `movement.jump` actor attribute as half the actor's strength score;
- configured `jump` movement type as hidden with a derived speed;
- filtered movement speeds in actor sheets and tooltips to skip hidden types.

Fyorl commented on module/canvas/ruler.mjs:
```diff
@@ -100,6 +100,11 @@ export default class TokenRuler5e extends foundry.canvas.placeables.tokens.Token
+    // If current action has a derived speed, use that instead
+    if ( CONFIG.DND5E.movementTypes[waypoint.action]?.deriveSpeed ) {
+      currActionSpeed = CONFIG.DND5E.movementTypes[waypoint.action].deriveSpeed(this.token);
+    }
```
I think this should be unnecessary now. We are retrieving the `currActionSpeed` from `movement[waypoint.action]` on line 95 which should give the correct result already since `movement.jump` exists now.

Aioros committed:
Clean up
